pkg/services/profiles/get.go

```diff
@@ -14,34 +15,80 @@ import (
 	pb "github.com/weaveworks/weave-gitops/pkg/api/profiles"
 )
 
-const (
-	wegoServiceName = "wego-app"
-	getProfilesPath = "/v1/profiles"
-)
-
 type GetOptions struct {
+	Name      string
+	Version   string
+	Cluster   string
 	Namespace string
-	ClientSet kubernetes.Interface
 	Writer    io.Writer
 	Port      string
 }
 
-func GetProfiles(ctx context.Context, opts GetOptions) error {
-	resp, err := kubernetesDoRequest(ctx, opts.Namespace, wegoServiceName, opts.Port, getProfilesPath, opts.ClientSet)
+// Get returns a list of available profiles.
+func (s *ProfilesSvc) Get(ctx context.Context, opts GetOptions) error {
+	profiles, err := doKubeGetRequest(ctx, opts.Namespace, wegoServiceName, opts.Port, getProfilesPath, s.ClientSet)
 	if err != nil {
 		return err
 	}
 
+	printProfiles(profiles, opts.Writer)
+
+	return nil
+}
+
+func doKubeGetRequest(ctx context.Context, namespace, serviceName, servicePort, path string, clientset kubernetes.Interface) (*pb.GetProfilesResponse, error) {
+	resp, err := kubernetesDoRequest(ctx, namespace, wegoServiceName, servicePort, getProfilesPath, clientset)
+	if err != nil {
+		return nil, err
+	}
+
 	profiles := &pb.GetProfilesResponse{}
 	err = jsonpb.UnmarshalString(string(resp), profiles)
 
 	if err != nil {
-		return fmt.Errorf("failed to unmarshal response: %w", err)
+		return nil, fmt.Errorf("failed to unmarshal response: %w", err)
 	}
+	return profiles, nil
+}
 
-	printProfiles(profiles, opts.Writer)
+// GetAvailableProfile returns a single available profile.
+func (s *ProfilesSvc) GetAvailableProfile(ctx context.Context, opts GetOptions) (*pb.Profile, error) {
```

Comment:
`Available` seems a bit odd, are you trying to differentiate the fact its not installed?